pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -311,6 +320,24 @@ func buildReplicationStreamSpec(
 		return nil, err
 	}
 
+	parts := int(streamMaxProcsPerPartition.Get(&evalCtx.Settings.SV))
+	const minPartitionSpans = 10
```

Comment:
just doesn't seem worth having a bunch of tiny procs? each proc needs to flush every X seconds to maintain RPO so if we split them up _too_ much we don't have much/anything writing to that buffer and end up flushing near-empty buffers. We want to group enough work together to make good use of a buffer, so if there isn't enough work to do that we still do that even after splitting it, we can just split splitting it?

Reply:
Yeah, i agree with these points, but I just find it hard to reason about how much "work" we think each span in a partition contains. I know that the BulkOracle will try to assign adjacent spans to the same node, allowing partitionSpans to merge adjacent ranges together. In other words, we like to make big spans. 

Suppose the replicating tenant has one big table with uniform traffic over the keyspace (e.g. kv0), can we predict the number partition spans we create as a function of the number of nodes? That might help us make more informed decisions around these constants. but also maybe i'm over thinking this.

Reply:
The streak affinity is only allowed if the skew between the node with the least assigned ranges (pre merging) is within a percentage of the ranges assigned to the current node. Early in partitioning, you’d thus be unable to make a large span since that span alone would skew the distribution. So you’d gradually make longer and longer streaks the more spans you partition, but you can’t have just eg every span go into one big partition. So if you have fewer than 10 merged spans, you just can’t have that many pre-merge spans